lib/client.js

```diff
@@ -1472,12 +1474,59 @@ function shouldSendContentLength (method) {
 }
 
 function write (client, request) {
+  const socket = client[kSocket]
+  const { body } = request
+
+  if (typeof body === 'function') {
+    // TODO (fix): What if socket is destroyed while
+    // waiting for body?
```

Comment:
If the factory is meant to be idempotent, destroying the body should be ok to do too.
We should advice to be careful about the state of the body